### DISCUSSION TITLE
Allow for x for multiplication in level 5 and up

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#838] [Language Ideas]
We see a lot of kids use or prefer x for multiplication. 

We now explain in level 5 straight away that * needs to be used, but that is hard and also somehow conveys the idea that programming is "weird" that we might want to push further along. 

So... Should we allow for the use of x, like :`5 x 6`? Do we allow both? Do we allow x but give a warning that is should really be *? Do we repair to *?

What do you think @MarleenGilsing? Would love toa also hear @OnnoEbbens's take after the level 5 lessons of the coming week.

**Comment by OnnoEbbens** [comment]
During my course there were no issues with people using the `x` instead of *. I think it really depends on the group. 

I would prefer not to allow the x for multiplication. I think level 5 is a good moment to learn to use * instead of x and you have to learn it anyway. A warning would be nice if it is clear that they try to use x instead of *.

**Comment by MarleenGilsing** [comment]
Do we see a lot of programs using x instead of *? Or do we just hear from children that x would be more natural? 
If we see the x problem often, maybe we should allow it at first. But I have a feeling that it doens't actually happen that often, and in that case I think it would be harder to unlearn using the x in a higher level than to start with the * right away.

**Reply by Felienne** [reply]
I have a feeling we see it often, but I can have a peek in the error dataset to confirm!

**Comment by Felienne** [comment]
Hi all! 

I looked at the data and it is common but not very common (a few hundred occurrences in all 1.5 million programs) so I guess we do not have to change the grammar to allow it. We could make a better error message, this is what you get in the current version:

![image](https://user-images.githubusercontent.com/1003685/139072876-2db44df5-a99c-432f-b376-afa8109518f4.png)

That clearly is not what we'd want, maybe we can solve this with [Confusables](https://pypi.org/project/confusables/) like we plan for ` versus '.
